Screens for editing transaction settings

vkjr opened the pull request:
fixes #21023 #21024

### Summary
Implemented screens for editing Max base fee, Priority fee, Nonce and Max gas amount.
This is only UI implementation and it is hidden under feature flag.
[status-go pr](https://github.com/status-im/status-go/pull/6124#pullrequestreview-2502899877) is not merged yet, so integration with status-go will be made separately


https://github.com/user-attachments/assets/a533c85d-557e-4a10-b254-589f65f7e856



### Review notes
[Figma designs](https://www.figma.com/design/xLs1KYmF4e6WwRTZVJKeUK/Wallet?node-id=17671-201810&m=dev)
In figma designs "max base fee" and "priority fee" use `token-input` component with `swap` button that allows switching between fiat and crypto. In my implementation there is no switching because it is not practical. 1 gwei is just a fraction of cent and user will never operate in fiat to set fees. Also we never operate with values less than 1 cent in other parts of our app. (cc @xAlisher)


### Testing notes

#### Platforms

- Android
- iOS

#### Areas that maybe impacted

##### Functional

- wallet / transactions

### Steps to test
- Open Status
- Start send flow, reach transaction confirmation screen
- Press transaction settings button, select "custom" menu and try different options


status: ready

vkjr committed:
Screens for editing transaction settings

status-im-auto commented:
### Jenkins Builds
<details>
<summary>Click to see older builds (24)</summary>

| :grey_question: | Commit | :hash: | Finished (UTC) | Duration | Platform | Result |
|-|-|-|-|-|-|-|
| :x: | 46862df6 | [#1](https://ci.status.im/job/status-mobile/job/prs/job/ios/job/PR-21838/1/) | 2024-12-17 15:39:32 | ~2 min | `ios` | [:page_facing_up:`log`](https://ci.status.im/job/status-mobile/job/prs/job/ios/job/PR-21838/1/consoleText) |
| :x: | 46862df6 | [#1](https://ci.status.im/job/status-mobile/job/prs/job/tests/job/PR-21838/1/) | 2024-12-17 15:41:31 | ~4 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-mobile/job/prs/job/tests/job/PR-21838/1/consoleText) |
| :heavy_check_mark: | 46862df6 | [#1](https://ci.status.im/job/status-mobile/job/prs/job/android-e2e/job/PR-21838/1/) | 2024-12-17 15:44:35 | ~8 min | `android-e2e` | [:robot:`apk`](https://status-im-mobile-prs.ams3.cdn.digitaloceanspaces.com/StatusIm-Mobile-241217-153632-46862d-pr21838-x86_64.apk) [:calling:](https://quickchart.io/qr?text=https%3A%2F%2Fstatus-im-mobile-prs.ams3.cdn.digitaloceanspaces.com%2FStatusIm-Mobile-241217-153632-46862d-pr21838-x86_64.apk&size=400x400&errorCorrectionLevel=L)|
| :heavy_check_mark: | 46862df6 | [#1](https://ci.status.im/job/status-mobile/job/prs/job/android/job/PR-21838/1/) | 2024-12-17 15:45:13 | ~8 min | `android` | [:robot:`apk`](https://status-im-mobile-prs.ams3.cdn.digitaloceanspaces.com/StatusIm-Mobile-241217-153637-46862d-pr21838-arm64-v8a.apk) [:calling:](https://quickchart.io/qr?text=https%3A%2F%2Fstatus-im-mobile-prs.ams3.cdn.digitaloceanspaces.com%2FStatusIm-Mobile-241217-153637-46862d-pr21838-arm64-v8a.apk&size=400x400&errorCorrectionLevel=L)|
| | | | | | | |
| :heavy_check_mark: | dd4e08a1 | [#2](https://ci.status.im/job/status-mobile/job/prs/job/tests/job/PR-21838/2/) | 2024-12-23 12:51:00 | ~4 min | `tests` | [:page_facing_up:`log`](https://status-im-mobile-prs.ams3.cdn.digitaloceanspaces.com/StatusIm-Mobile-241223-124625-dd4e08-pr21838-tests.log) |
| :heavy_check_mark: | dd4e08a1 | [#2](https://ci.status.im/job/status-mobile/job/prs/job/android-e2e/job/PR-21838/2/) | 2024-12-23 12:53:00 | ~6 min | `android-e2e` | [:robot:`apk`](https://status-im-mobile-prs.ams3.cdn.digitaloceanspaces.com/StatusIm-Mobile-241223-124621-dd4e08-pr21838-x86_64.apk) [:calling:](https://quickchart.io/qr?text=https%3A%2F%2Fstatus-im-mobile-prs.ams3.cdn.digitaloceanspaces.com%2FStatusIm-Mobile-241223-124621-dd4e08-pr21838-x86_64.apk&size=400x400&errorCorrectionLevel=L)|
| :heavy_check_mark: | dd4e08a1 | [#2](https://ci.status.im/job/status-mobile/job/prs/job/ios/job/PR-21838/2/) | 2024-12-23 12:53:17 | ~6 min | `ios` | [:iphone:`ipa`](https://i.diawi.com/3ENXnU) [:calling:](https://quickchart.io/qr?text=https%3A%2F%2Fi.diawi.com%2F3ENXnU&size=400x400&errorCorrectionLevel=L)|
| :heavy_check_mark: | dd4e08a1 | [#2](https://ci.status.im/job/status-mobile/job/prs/job/android/job/PR-21838/2/) | 2024-12-23 12:54:44 | ~8 min | `android` | [:robot:`apk`](https://status-im-mobile-prs.ams3.cdn.digitaloceanspaces.com/StatusIm-Mobile-241223-124621-dd4e08-pr21838-arm64-v8a.apk) [:calling:](https://quickchart.io/qr?text=https%3A%2F%2Fstatus-im-mobile-prs.ams3.cdn.digitaloceanspaces.com%2FStatusIm-Mobile-241223-124621-dd4e08-pr21838-arm64-v8a.apk&size=400x400&errorCorrectionLevel=L)|
| | | | | | | |
| :heavy_check_mark: | 31c80d21 | [#3](https://ci.status.im/job/status-mobile/job/prs/job/tests/job/PR-21838/3/) | 2025-01-06 12:54:56 | ~4 min | `tests` | [:page_facing_up:`log`](https://status-im-mobile-prs.ams3.cdn.digitaloceanspaces.com/StatusIm-Mobile-250106-125040-31c80d-pr21838-tests.log) |
| :heavy_check_mark: | 31c80d21 | [#3](https://ci.status.im/job/status-mobile/job/prs/job/ios/job/PR-21838/3/) | 2025-01-06 12:57:35 | ~6 min | `ios` | [:iphone:`ipa`](https://i.diawi.com/saPRx7) [:calling:](https://quickchart.io/qr?text=https%3A%2F%2Fi.diawi.com%2FsaPRx7&size=400x400&errorCorrectionLevel=L)|
| :heavy_check_mark: | 31c80d21 | [#3](https://ci.status.im/job/status-mobile/job/prs/job/android-e2e/job/PR-21838/3/) | 2025-01-06 12:58:07 | ~7 min | `android-e2e` | [:robot:`apk`](https://status-im-mobile-prs.ams3.cdn.digitaloceanspaces.com/StatusIm-Mobile-250106-125036-31c80d-pr21838-x86_64.apk) [:calling:](https://quickchart.io/qr?text=https%3A%2F%2Fstatus-im-mobile-prs.ams3.cdn.digitaloceanspaces.com%2FStatusIm-Mobile-250106-125036-31c80d-pr21838-x86_64.apk&size=400x400&errorCorrectionLevel=L)|
| :heavy_check_mark: | 31c80d21 | [#3](https://ci.status.im/job/status-mobile/job/prs/job/android/job/PR-21838/3/) | 2025-01-06 12:58:46 | ~8 min | `android` | [:robot:`apk`](https://status-im-mobile-prs.ams3.cdn.digitaloceanspaces.com/StatusIm-Mobile-250106-125036-31c80d-pr21838-arm64-v8a.apk) [:calling:](https://quickchart.io/qr?text=https%3A%2F%2Fstatus-im-mobile-prs.ams3.cdn.digitaloceanspaces.com%2FStatusIm-Mobile-250106-125036-31c80d-pr21838-arm64-v8a.apk&size=400x400&errorCorrectionLevel=L)|
| | | | | | | |
| :heavy_check_mark: | c5a2efde | [#4](https://ci.status.im/job/status-mobile/job/prs/job/tests/job/PR-21838/4/) | 2025-01-07 15:06:32 | ~5 min | `tests` | [:page_facing_up:`log`](https://status-im-mobile-prs.ams3.cdn.digitaloceanspaces.com/StatusIm-Mobile-250107-150126-c5a2ef-pr21838-tests.log) |
| :heavy_check_mark: | c5a2efde | [#4](https://ci.status.im/job/status-mobile/job/prs/job/ios/job/PR-21838/4/) | 2025-01-07 15:08:29 | ~7 min | `ios` | [:iphone:`ipa`](https://i.diawi.com/pR5BoZ) [:calling:](https://quickchart.io/qr?text=https%3A%2F%2Fi.diawi.com%2FpR5BoZ&size=400x400&errorCorrectionLevel=L)|
| :heavy_check_mark: | c5a2efde | [#4](https://ci.status.im/job/status-mobile/job/prs/job/android-e2e/job/PR-21838/4/) | 2025-01-07 15:10:44 | ~9 min | `android-e2e` | [:robot:`apk`](https://status-im-mobile-prs.ams3.cdn.digitaloceanspaces.com/StatusIm-Mobile-250107-150122-c5a2ef-pr21838-x86_64.apk) [:calling:](https://quickchart.io/qr?text=https%3A%2F%2Fstatus-im-mobile-prs.ams3.cdn.digitaloceanspaces.com%2FStatusIm-Mobile-250107-150122-c5a2ef-pr21838-x86_64.apk&size=400x400&errorCorrectionLevel=L)|
| :heavy_check_mark: | c5a2efde | [#4](https://ci.status.im/job/status-mobile/job/prs/job/android/job/PR-21838/4/) | 2025-01-07 15:11:08 | ~9 min | `android` | [:robot:`apk`](https://status-im-mobile-prs.ams3.cdn.digitaloceanspaces.com/StatusIm-Mobile-250107-150126-c5a2ef-pr21838-arm64-v8a.apk) [:calling:](https://quickchart.io/qr?text=https%3A%2F%2Fstatus-im-mobile-prs.ams3.cdn.digitaloceanspaces.com%2FStatusIm-Mobile-250107-150126-c5a2ef-pr21838-arm64-v8a.apk&size=400x400&errorCorrectionLevel=L)|
| | | | | | | |
| :heavy_check_mark: | 34177a65 | [#5](https://ci.status.im/job/status-mobile/job/prs/job/tests/job/PR-21838/5/) | 2025-01-07 17:57:13 | ~4 min | `tests` | [:page_facing_up:`log`](https://status-im-mobile-prs.ams3.cdn.digitaloceanspaces.com/StatusIm-Mobile-250107-175234-34177a-pr21838-tests.log) |
| :heavy_check_mark: | 34177a65 | [#5](https://ci.status.im/job/status-mobile/job/prs/job/android-e2e/job/PR-21838/5/) | 2025-01-07 17:59:34 | ~7 min | `android-e2e` | [:robot:`apk`](https://status-im-mobile-prs.ams3.cdn.digitaloceanspaces.com/StatusIm-Mobile-250107-175230-34177a-pr21838-x86_64.apk) [:calling:](https://quickchart.io/qr?text=https%3A%2F%2Fstatus-im-mobile-prs.ams3.cdn.digitaloceanspaces.com%2FStatusIm-Mobile-250107-175230-34177a-pr21838-x86_64.apk&size=400x400&errorCorrectionLevel=L)|
| :heavy_check_mark: | 34177a65 | [#5](https://ci.status.im/job/status-mobile/job/prs/job/android/job/PR-21838/5/) | 2025-01-07 18:00:52 | ~8 min | `android` | [:robot:`apk`](https://status-im-mobile-prs.ams3.cdn.digitaloceanspaces.com/StatusIm-Mobile-250107-175230-34177a-pr21838-arm64-v8a.apk) [:calling:](https://quickchart.io/qr?text=https%3A%2F%2Fstatus-im-mobile-prs.ams3.cdn.digitaloceanspaces.com%2FStatusIm-Mobile-250107-175230-34177a-pr21838-arm64-v8a.apk&size=400x400&errorCorrectionLevel=L)|
| :heavy_check_mark: | 34177a65 | [#5](https://ci.status.im/job/status-mobile/job/prs/job/ios/job/PR-21838/5/) | 2025-01-07 18:02:31 | ~10 min | `ios` | [:iphone:`ipa`](https://i.diawi.com/de7C9x) [:calling:](https://quickchart.io/qr?text=https%3A%2F%2Fi.diawi.com%2Fde7C9x&size=400x400&errorCorrectionLevel=L)|
| | | | | | | |
| :heavy_check_mark: | a79acef0 | [#6](https://ci.status.im/job/status-mobile/job/prs/job/tests/job/PR-21838/6/) | 2025-01-09 14:26:49 | ~5 min | `tests` | [:page_facing_up:`log`](https://status-im-mobile-prs.ams3.cdn.digitaloceanspaces.com/StatusIm-Mobile-250109-142141-a79ace-pr21838-tests.log) |
| :heavy_check_mark: | a79acef0 | [#6](https://ci.status.im/job/status-mobile/job/prs/job/ios/job/PR-21838/6/) | 2025-01-09 14:29:09 | ~7 min | `ios` | [:iphone:`ipa`](https://i.diawi.com/rkN9Uj) [:calling:](https://quickchart.io/qr?text=https%3A%2F%2Fi.diawi.com%2FrkN9Uj&size=400x400&errorCorrectionLevel=L)|
| :heavy_check_mark: | a79acef0 | [#6](https://ci.status.im/job/status-mobile/job/prs/job/android-e2e/job/PR-21838/6/) | 2025-01-09 14:29:20 | ~7 min | `android-e2e` | [:robot:`apk`](https://status-im-mobile-prs.ams3.cdn.digitaloceanspaces.com/StatusIm-Mobile-250109-142137-a79ace-pr21838-x86_64.apk) [:calling:](https://quickchart.io/qr?text=https%3A%2F%2Fstatus-im-mobile-prs.ams3.cdn.digitaloceanspaces.com%2FStatusIm-Mobile-250109-142137-a79ace-pr21838-x86_64.apk&size=400x400&errorCorrectionLevel=L)|
| :heavy_check_mark: | a79acef0 | [#6](https://ci.status.im/job/status-mobile/job/prs/job/android/job/PR-21838/6/) | 2025-01-09 14:29:51 | ~8 min | `android` | [:robot:`apk`](https://status-im-mobile-prs.ams3.cdn.digitaloceanspaces.com/StatusIm-Mobile-250109-142137-a79ace-pr21838-arm64-v8a.apk) [:calling:](https://quickchart.io/qr?text=https%3A%2F%2Fstatus-im-mobile-prs.ams3.cdn.digitaloceanspaces.com%2FStatusIm-Mobile-250109-142137-a79ace-pr21838-arm64-v8a.apk&size=400x400&errorCorrectionLevel=L)|
</details>

| :grey_question: | Commit | :hash: | Finished (UTC) | Duration | Platform | Result |
|-|-|-|-|-|-|-|
| :heavy_check_mark: | a1a312f6 | [#7](https://ci.status.im/job/status-mobile/job/prs/job/tests/job/PR-21838/7/) | 2025-01-10 16:53:56 | ~4 min | `tests` | [:page_facing_up:`log`](https://status-im-mobile-prs.ams3.cdn.digitaloceanspaces.com/StatusIm-Mobile-250110-164911-a1a312-pr21838-tests.log) |
| :heavy_check_mark: | a1a312f6 | [#7](https://ci.status.im/job/status-mobile/job/prs/job/android-e2e/job/PR-21838/7/) | 2025-01-10 16:55:41 | ~6 min | `android-e2e` | [:robot:`apk`](https://status-im-mobile-prs.ams3.cdn.digitaloceanspaces.com/StatusIm-Mobile-250110-164907-a1a312-pr21838-x86_64.apk) [:calling:](https://quickchart.io/qr?text=https%3A%2F%2Fstatus-im-mobile-prs.ams3.cdn.digitaloceanspaces.com%2FStatusIm-Mobile-250110-164907-a1a312-pr21838-x86_64.apk&size=400x400&errorCorrectionLevel=L)|
| :heavy_check_mark: | a1a312f6 | [#7](https://ci.status.im/job/status-mobile/job/prs/job/android/job/PR-21838/7/) | 2025-01-10 16:57:43 | ~8 min | `android` | [:robot:`apk`](https://status-im-mobile-prs.ams3.cdn.digitaloceanspaces.com/StatusIm-Mobile-250110-164907-a1a312-pr21838-arm64-v8a.apk) [:calling:](https://quickchart.io/qr?text=https%3A%2F%2Fstatus-im-mobile-prs.ams3.cdn.digitaloceanspaces.com%2FStatusIm-Mobile-250110-164907-a1a312-pr21838-arm64-v8a.apk&size=400x400&errorCorrectionLevel=L)|
| :heavy_check_mark: | a1a312f6 | [#7](https://ci.status.im/job/status-mobile/job/prs/job/ios/job/PR-21838/7/) | 2025-01-10 16:58:22 | ~9 min | `ios` | [:iphone:`ipa`](https://i.diawi.com/Y3PehC) [:calling:](https://quickchart.io/qr?text=https%3A%2F%2Fi.diawi.com%2FY3PehC&size=400x400&errorCorrectionLevel=L)|
| | | | | | | |
| :heavy_check_mark: | 9d18cf9a | [#8](https://ci.status.im/job/status-mobile/job/prs/job/tests/job/PR-21838/8/) | 2025-01-20 11:13:58 | ~5 min | `tests` | [:page_facing_up:`log`](https://status-im-mobile-prs.ams3.cdn.digitaloceanspaces.com/StatusIm-Mobile-250120-110807-9d18cf-pr21838-tests.log) |
| :heavy_check_mark: | 9d18cf9a | [#8](https://ci.status.im/job/status-mobile/job/prs/job/android-e2e/job/PR-21838/8/) | 2025-01-20 11:18:24 | ~10 min | `android-e2e` | [:robot:`apk`](https://status-im-mobile-prs.ams3.cdn.digitaloceanspaces.com/StatusIm-Mobile-250120-110802-9d18cf-pr21838-x86_64.apk) [:calling:](https://quickchart.io/qr?text=https%3A%2F%2Fstatus-im-mobile-prs.ams3.cdn.digitaloceanspaces.com%2FStatusIm-Mobile-250120-110802-9d18cf-pr21838-x86_64.apk&size=400x400&errorCorrectionLevel=L)|
| :heavy_check_mark: | 9d18cf9a | [#8](https://ci.status.im/job/status-mobile/job/prs/job/ios/job/PR-21838/8/) | 2025-01-20 11:18:41 | ~10 min | `ios` | [:iphone:`ipa`](https://i.diawi.com/StGUtH) [:calling:](https://quickchart.io/qr?text=https%3A%2F%2Fi.diawi.com%2FStGUtH&size=400x400&errorCorrectionLevel=L)|
| :heavy_check_mark: | 9d18cf9a | [#8](https://ci.status.im/job/status-mobile/job/prs/job/android/job/PR-21838/8/) | 2025-01-20 11:18:47 | ~10 min | `android` | [:robot:`apk`](https://status-im-mobile-prs.ams3.cdn.digitaloceanspaces.com/StatusIm-Mobile-250120-110802-9d18cf-pr21838-arm64-v8a.apk) [:calling:](https://quickchart.io/qr?text=https%3A%2F%2Fstatus-im-mobile-prs.ams3.cdn.digitaloceanspaces.com%2FStatusIm-Mobile-250120-110802-9d18cf-pr21838-arm64-v8a.apk&size=400x400&errorCorrectionLevel=L)|

vkjr commented on src/quo/components/inputs/input/view.cljs:
```diff
@@ -7,8 +7,8 @@
 
-(defn- label-&-counter
-  [{:keys [label current-chars char-limit variant-colors theme]}]
+(defn- label-line
```
In designs input field has labels at top-left and top-right sides. In our component we had only characters counter in the right side, so i added a right-side label.

vkjr commented on src/quo/components/tags/network_tags/view.cljs:
```diff
@@ -14,14 +14,16 @@
-       :number (count networks)
-       :size   :size-16}
-      networks]
+     (when networks
```
If `networks` is empty we don't need this component because it eats some space and leads to incorrect alignment of text after it.

vkjr commented on src/quo/components/wallet/token_input/view.cljs:
```diff
@@ -20,17 +20,18 @@
-  (let [theme        (quo.theme/use-theme)
-        window-width (:width (rn/get-window))]
+  [{:keys [token-symbol on-token-press value error? on-swap currency-symbol show-token-icon?
+           swappable?]}]
```
Two new features of `token-input`:
If not `swappable?` we won't display the button to swap between crypto and fiat values.
If not `show-token-icon?` we do not show icon. Because `GWEI` or `UNITS` that we enter in transaction settings screens do not have icons.

vkjr commented on src/status_im/common/controlled_input/utils.cljs:
```diff
@@ -78,16 +78,19 @@
 (defn set-upper-limit
   [state limit]
-  (when limit
+  (if limit
```
Without this change `set-upper-limit` breaks the usage of `->` because no value returned when limit not applied

vkjr commented on src/status_im/contexts/wallet/send/events.cljs:
```diff
@@ -768,3 +768,41 @@
             (not keep-tx-data?) (update-in [:wallet :ui :send] dissoc :tx-type))
       :fx [[:dispatch [:navigate-back]]]})))
+
+(rf/reg-event-fx
```
I do not know yet in what form we will get the recommended and current values for transaction settings from `status-go`, so for now I've stored test values like that. Not a final form I believe.

vkjr commented on src/status_im/navigation/events.cljs:
```diff
@@ -30,8 +30,9 @@
-  {:db (all-screens-params db (first comp-id) screen-params)
-   :fx [[:navigate-to-within-stack (conj comp-id (:theme db))]]})
+  {:db         (all-screens-params db (first comp-id) screen-params)
+   :dispatch-n [[:hide-bottom-sheet]]
```
Bottom sheets should be hidden when navigate to another screen. Copied from `:navigate-to` event above in the code.

clauxx replied:
`:dispatch-n` is redundant here. Instead we can put `[:dispatch [:hide-bottom-sheet]]` inside `:fx` before navigating instead.

mohsen-ghafouri commented on src/status_im/contexts/wallet/send/transaction_settings/view.cljs:
```diff
@@ -106,3 +130,91 @@
        :button-one-props {:on-press #(rf/dispatch [:hide-bottom-sheet])}
        :button-one-label (i18n/label :t/confirm)}]]))
+
+
+
```
extra lines

vkjr committed:
extra lines removed

status-im-auto commented:
## 100% of end-end tests have passed
```
Total executed tests: 8
Failed tests: 0
Expected to fail tests: 0
Passed tests: 8
```
<h3>Passed tests (8)</h3><details><summary>Click to expand</summary><br/><br/><h4>Class TestCommunityOneDeviceMerged:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3762278">test_restore_multiaccount_with_waku_backup_remove_profile_switch</a>, id: 703133 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTES0FXTEXbAHvg74YdI">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3762275">test_community_copy_and_paste_message_in_chat_input</a>, id: 702742 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTES0FXTEXbAHvg74YdI">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestWalletOneDevice:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3762281">test_wallet_add_remove_regular_account</a>, id: 727231 </b></td></tr><tr><td></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3762282">test_wallet_balance_mainnet</a>, id: 740490 </b></td></tr><tr><td></td></tr></tbody></table><h4>Class TestWalletMultipleDevice:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3762280">test_wallet_send_asset_from_drawer</a>, id: 727230 </b></td></tr><tr><td></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3762279">test_wallet_send_eth</a>, id: 727229 </b></td></tr><tr><td></td></tr></tbody></table><h4>Class TestOneToOneChatMultipleSharedDevicesNewUi:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3762276">test_1_1_chat_non_latin_messages_stack_update_profile_photo</a>, id: 702745 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESRqSvTyyC5QFhkTgn">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESCyKHC4y4A3NMQBw6">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestCommunityMultipleDeviceMerged:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3762277">test_community_message_edit</a>, id: 702843 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESY2Q9NRQS4bwAp45a">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTES6GfYFYe3I7dZPws8">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table></details>

clauxx commented on src/quo/components/inputs/input/view.cljs:
```diff
@@ -18,17 +18,25 @@
         :size   :paragraph-2}
-       count-text]])])
+       label-right]])
+   (when char-limit
+     (when-let [count-text (some->> char-limit
+                                    (str current-chars "/"))]
```
do we need all 3 conditions (when, when-let, some->>)? I guess if we just keep the `when`, we no longer will need the `when-let` and `some->>`, cause `char-limit` will be always defined inside.

clauxx commented on src/status_im/contexts/wallet/send/transaction_settings/view.cljs:
```diff
@@ -2,53 +2,68 @@
+        nonce          (:current (rf/sub [:wallet/tx-settings-nonce]))]
+    [rn/view
+     [quo/drawer-top
+      {:title "Custom"}]
```
Should be translatable

clauxx commented on src/status_im/contexts/wallet/send/transaction_settings/view.cljs:
```diff
@@ -58,18 +73,27 @@
-                    :label             :text
-                    :preview-size      :size-32}
-                   {:title             "Fast ~40s"
+       :data      [{:title (str (i18n/label :t/normal) "~60s")
+                    :image-props
+                    "🍿"
+                    :description-props
+                    {:text "€1.45"}
+                    :image
+                    :emoji
+                    :description
```
the formatting is strange here 🤔

vkjr commented:
@clauxx, thanks for the comments!

vkjr committed:
review notes fixed

pavloburykh commented:
@vkjr thank you for the PR! Got a few questions/issues


### QUESTION 1 

Should Custom settings be applied for Bridge flow? Currently they are applied.

![photo_2025-01-10 17 40 20](https://github.com/user-attachments/assets/32be1215-0f86-4fba-a411-712b4d642acb)

pavloburykh commented:
### QUESTION 2 

Crypto/Fiat amount should be calculated in Max base fee and Priority fee according to Figma. Currently it is not calculated.
Also, Priority fee has a button for switching to Fiat/Crypto. 

Is it waiting for status go integration as well or this is a bug? See screenshots from Figma below. 

<img width="284" alt="1" src="https://github.com/user-attachments/assets/66dc9b1f-4799-41f8-847a-344dada10359" />

<img width="274" alt="Wallet – Figma 2025-01-10 17-51-55" src="https://github.com/user-attachments/assets/04d07f99-6603-4a2b-90bb-2b1175aba207" />

pavloburykh commented:
### QUESTION 3

Read more option does not open anything. Also I haven't find anything in designs. Should it be implemented in future? 


https://github.com/user-attachments/assets/f513bacf-1262-4a42-bc3e-ffee443d6d29

pavloburykh commented:
### QUESTION 4

How to get Max gas amount error? Trying to enter highest possible number but still no error. Or we should wait for backend test it? 

**Actual result:** 
![photo_2025-01-10 18 02 38](https://github.com/user-attachments/assets/4d70fc79-8afd-4fe8-80a4-432e94df96dc)


 **Expected result:** 
<img width="255" alt="Wallet – Figma 2025-01-10 18-12-51" src="https://github.com/user-attachments/assets/6540ad42-9e46-4371-b43d-d911c2b4bd87" />

pavloburykh commented:
### ISSUE 1 Part of text is missing in Max base fee and Priority fee info screens

**Expected result:** 
<img width="301" alt="Wallet – Figma 2025-01-10 17-56-46" src="https://github.com/user-attachments/assets/72dc8a39-6413-4f14-b07f-395ac67505f5" />
<img width="301" alt="Wallet – Figma 2025-01-10 17-56-46" src="https://github.com/user-attachments/assets/fd844763-2197-402c-bcd5-03a56d42bf3f" />


**Actual result:** 
![photo_2025-01-10 17 59 29](https://github.com/user-attachments/assets/d2e1ad9a-d076-4193-a9ae-cbc9d350187f)
![photo_2025-01-10 17 59 32](https://github.com/user-attachments/assets/932e9161-4457-45e2-af0a-0978d91bc777)

vkjr commented:
@pavloburykh, thanks for carefully checking this one!
My bad, I should have written somewhere in the comments that designs for this were refined and updated just yesterday. 
It was yesterday when we added new descriptions, changed the upper/higher limits, etc. So I suggest to keep those changes out of scope of this PR? I will create a new issue and will move all your findings there, wdyt? 
 
Thanks for your finding regarding `bridge`. I will check with @saledjenic if custom settings are applicable for bridge transactions.

pavloburykh commented:
>  I will create a new issue and will move all your findings there, wdyt?

Sounds good @vkjr! I am moving current PR forward.

saledjenic commented:
> I will check with @saledjenic if custom settings are applicable for bridge transactions.

@vkjr @pavloburykh custom fees/nonce/gas amount is applicable for every single tx. Either it's an approval tx or a regular tx.

For example, the current UX (at least in the desktop app) in case approval is needed is like this:
- for swap:
  - the user is about to send approval tx (at this point customization for approval tx can be applied), approval tx sent
  - the swap tx, can be placed after approval tx is fully executed, then the user goes through a similar sending flow where customization for swap tx can be applied
- for bridge:
  - approval and bridge txs are placed at the same time (cause no technical limitations like we have for swap) and not sure how that will be solved (not sure if that's designed, if it is, then it's not implemented yet), will we have a single bridge tx popup for review where the user will at the same time set custom params for both (approval and bridge) txs, or we will align that with the swap flow. That's more a question for @benjthayer but from the technical side of things, backend logic for setting custom params for all txs is already in place.
